.github/workflows/main.yml

```diff
@@ -11,61 +11,45 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.9'
-      - name: Install project
-        run: pip install .
-      - name: Run command
+          python-version: '3.11'
+      - name: Check formatting
         run: |
-          pip3 install riot
+          python -m pip install riot==0.18.0
           riot -v run -s black -- --check .
   mypy:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.9'
+          python-version: '3.11'
       - name: Install project
         run: pip install .
       - name: Run command
         run: |
-          pip3 install riot
+          python -m pip install riot==0.18.0
           riot -v run -s mypy
   flake8:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: '3.9'
-      - name: Install project
-        run: pip install .
+          python-version: '3.11'
       - name: Run command
         run: |
-          pip3 install riot
+          python -m pip install riot==0.18.0
           riot -v run -s flake8
   test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10", pypy-3.7]
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
-      - name: Setup Python 3 (for riot)
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
         with:
-          python-version: "3.9"
-      - name: Install project
-        run: |
-          pip install setuptools
-          pip install .
-      - name: Run tests
-        run: |
-          python3.9 -m pip install riot
-          riot -v run --python=${{ matrix.python-version }} smoke-test
-          riot -v run --python=${{ matrix.python-version }} tests
+          fetch-depth: 0
+      - run: |
+          docker run \
+            --rm \
+            -v $PWD:/work/envier \
+            -w /work/envier \
+            ghcr.io/kyle-verhoog/envier/test:a0e8d6f38400e8bf21fc0e30ccba4001a6885884 \
```

Comment:
will switch over to the image published in https://github.com/DataDog/envier/pull/25 after that PR is merged.